### PR TITLE
Fix some memory leaks

### PR DESCRIPTION
**Pull request by Einh06**
**What is in this PR**
This PR fixes couple of memory leaks, mostly due to retain cycle in requests.

NOTE: Requires a release of wire-ios-mocktransport

### CODE REVIEW

**Comment by marcoconti83** [comment]
Here's the PR on wire-ios-mocktransport: https://github.com/wireapp/wire-ios-mocktransport/pull/4

File: zmessaging-cocoa.xcodeproj/xcshareddata/xcschemes/zmessaging-ios.xcscheme [hunk 1]
```diff
@@ -116,7 +116,7 @@
       </AdditionalOptions>
    </LaunchAction>
    <ProfileAction
-      buildConfiguration = "Release"
```
Comment (marcoconti83) [inline zmessaging-cocoa.xcodeproj/xcshareddata/xcschemes/zmessaging-ios.xcscheme]: Is this change legit?

**Comment by marcoconti83** [comment]
Nice job removing the retain cycles! Did you notice any improvement in the speed of tests?

File: zmessaging-cocoa.xcodeproj/xcshareddata/xcschemes/zmessaging-ios.xcscheme (hunk 1 above)
Comment (Einh06) [inline zmessaging-cocoa.xcodeproj/xcshareddata/xcschemes/zmessaging-ios.xcscheme]: Yes it is. To be able to profile and have the symbols, i needed to profile in debug mode.

**Comment by Einh06** [comment]
I haven't seen much improvement in regard of tests speed. Further research need to be done to understand why our tests take so long.

File: zmessaging-cocoa.xcodeproj/xcshareddata/xcschemes/zmessaging-ios.xcscheme (hunk 1 above)
Comment (marcoconti83) [inline zmessaging-cocoa.xcodeproj/xcshareddata/xcschemes/zmessaging-ios.xcscheme]: Profiling in Develop though does not give you relevant information on what's happening in Release

**Pull request merged**